File: 自写/UI/ImagePicker&CropDemo/ImagePicker&CropDemo/ViewController.m
//
//  ViewController.m
//  ImagePicker&CropDemo
//
//  Created by luozhijun on 14-12-12.
//  Copyright (c) 2014年 luozhijun. All rights reserved.
//

#import "ViewController.h"
#import "UIViewController+imagePicker.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *targetImageView;
@property (weak, nonatomic) IBOutlet UITextView *textView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    

    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillShow:) name:UIKeyboardWillShowNotification object:nil];
    [self.targetImageView addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(imageViewTapped)]];
}

- (void)imageViewTapped
{
    // jepg图片压缩质量
    self.compressionQuality = @0.00001;
    
    self.editingImageView = self.targetImageView;
    self.imageCropMode = @(UIViewControllerImagePickerCropModeCircle);
    
    [self showImagePickerActionSheet];
}

- (void)keyboardWillShow:(NSNotification *)note
{
    NSLog(@"%@", note);
}

@end

File: 自写/Demo/aaaaaa/aaaaaa/ViewController.swift
//
//  ViewController.swift
//  aaaaaa
//
//  Created by luozhijun on 2017/3/23.
//  Copyright © 2017年 RickLuo. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let value = "哈哈哈".encryptedString(with: .SHA512, isHmac: true, hmacKey: "tzVIhxznsUR2")
        print(value)
        print(value.1)
    }
}
